### ISSUE TITLE
`valid-types` doesn't allow users to specify the properties of array items

### ISSUE THREAD

**Issue by gurpreetatwal** [gajus/eslint-plugin-jsdoc#596]
## Expected behavior
`valid-types` should allow users to specify the properties of values in an array as documented on the JSDoc website: https://jsdoc.app/tags-param.html#parameters-with-properties

## Actual behavior
```
  6:0  error  Syntax error in namepath: employees[].name        jsdoc/valid-types
  7:0  error  Syntax error in namepath: employees[].department  jsdoc/valid-types
```

## ESLint Config

```js
'use strict';

module.exports = {
  env: {
    node: true,
  },
  plugins: ['jsdoc'],
  rules: {
    'jsdoc/valid-types': [
      'error',
      { allowEmptyNamepaths: true, checkSeesForNamepaths: false },
    ],
  },
};
```

## ESLint sample

```js
'use strict';

/**
 * Assign the project to a list of employees.
 * @param {Object[]} employees - The employees who are responsible for the project.
 * @param {string} employees[].name - The name of an employee.
 * @param {string} employees[].department - The employee's department.
 */
function assign(employees) {
  // ...
}
```

## Environment

* Node version: v12.18.1
* ESLint version: v7.3.1
* `eslint-plugin-jsdoc` version: v28.5.1

**Comment by brettz9** [comment]
This is actually due to a jsdoctypeparser issue. We need to have such `catharsis` parser rules (as used by jsdoc) moved over to `jsdoctypeparser` to allow for these form of namepaths (though excluding this from typescript mode, I believe, and possibly also "closure" mode). (The reason `check-param-names` handles this format is because we do our own (incomplete) name parsing in that rule, but this really needs to be fixed in jsdoctypeparser.)

This is blocked by https://github.com/jsdoctypeparser/jsdoctypeparser/issues/117 so feel free to subscribe there. It may require some tweaking in `eslint-plugin-jsdoc` as well (at least if we separate name from namepath parsing as per https://github.com/jsdoctypeparser/jsdoctypeparser/issues/117 ) so keeping this issue open for now.

**Comment by brettz9** [comment]
This is no longer reporting errors, so added a test and closing.

I've just pushed a fix to catch some errors along these lines (e.g., `employees[.name` with a missing bracket now errs out). However, #374 is being used to track a potentially more proper namepath (or name) parsing.
